### PR TITLE
1.0, 1.1 template tests

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs [hunk 1]
```diff
@@ -34,6 +34,52 @@ public TemplateFunctionalTests(AzureFixture fixture, ITestOutputHelper outputHel
...
+
+                using (var httpClient = site.CreateClient())
+                {
+                    var getResult = await httpClient.GetAsync("/");
+                    getResult.EnsureSuccessStatusCode();
+                    Assert.Contains(expected, await getResult.Content.ReadAsStringAsync());
+
+                    getResult = await httpClient.GetAsync("/runtimeInfo");
+                    getResult.EnsureSuccessStatusCode();
+                }
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: I think you forgot to verify the runtime info

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs [hunk 2]
```diff
@@ -34,6 +34,93 @@ public TemplateFunctionalTests(AzureFixture fixture, ITestOutputHelper outputHel
...
+                var publishDirectory = testDirectory.CreateSubdirectory("publish");
+                var dotnet = DotNet(logger, testDirectory, "1.1");
+
+                await dotnet.ExecuteAndAssertAsync("new " + template);
+
+                UpdateCSProj(testDirectory, $"Templates\\Legacy.{expectedRuntime}.{template}.csproj");
+
+                InjectMiddlware(testDirectory, RuntimeInformationMiddlewareType, RuntimeInformationMiddlewareFile);
+
+                await dotnet.ExecuteAndAssertAsync("restore -s https://api.nuget.org/v3/index.json");
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: Maybe comment why

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs [hunk 3]
```diff
@@ -34,6 +34,93 @@ public TemplateFunctionalTests(AzureFixture fixture, ITestOutputHelper outputHel
...
+        private void ValidateLegacyRuntimeInfo(RuntimeInfo runtimeInfo, DirectoryInfo publishPath, string expectedRuntime, string dotnetPath)
+        {
+            var cacheAssemblies = new HashSet<string>(File.ReadAllLines("DotNetCache.txt"), StringComparer.InvariantCultureIgnoreCase);
+            var appModules = PathUtilities.GetAllModules(publishPath);
+            var runtimeModules = PathUtilities.GetLatestSharedRuntimeAssemblies(dotnetPath, out var runtimeVersion);
+
+            foreach (var runtimeInfoModule in runtimeInfo.Modules)
+            {
+                //// Check if assembly that is in the cache is loaded from it
+                //if (cacheAssemblies.Contains(Path.GetFileNameWithoutExtension(runtimeInfoModule.ModuleName)))
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: Whats the deal here?

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+Microsoft.AspNetCore.Authentication.Cookies
+Microsoft.AspNetCore.Authentication.Facebook
+Microsoft.AspNetCore.Authentication.Google
+Microsoft.AspNetCore.Authentication.JwtBearer
+Microsoft.AspNetCore.Authentication.MicrosoftAccount
+Microsoft.AspNetCore.Authentication.OAuth
+Microsoft.AspNetCore.Authentication.OpenIdConnect
+Microsoft.AspNetCore.Authentication.Twitter
+Microsoft.AspNetCore.Authorization
+microsoft.aspnetcore.azureappservicesintegration
```
Comment (muratg) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt]: Why casing difference? :)

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs [hunk 5]
```diff
@@ -34,6 +34,91 @@ public TemplateFunctionalTests(AzureFixture fixture, ITestOutputHelper outputHel
...
+
+                var dotnet = DotNet(logger, testDirectory, "1.1");
+
+                await dotnet.ExecuteAndAssertAsync("new " + template);
+
+                UpdateCSProj(testDirectory, $"Templates\\Legacy.{expectedRuntime}.{template}.csproj");
+
+                InjectMiddlware(testDirectory, RuntimeInformationMiddlewareType, RuntimeInformationMiddlewareFile);
+
+                var pp = await site.GetPublishingProfileAsync();
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: name

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Templates/AppServicesWithSiteExtensions.json [hunk 6]
```diff
@@ -29,22 +29,23 @@
...
-            "version": "[parameters('extensionVersion')]",
-            "feed_url": "[parameters('extensionFeed')]"
-          },
-          "dependsOn": [
-            "[resourceId('Microsoft.Web/Sites', parameters('siteName'))]"
-          ]
-        }
-      ]
+      }
+      //,
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Templates/AppServicesWithSiteExtensions.json]: ?

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs [hunk 7]
```diff
@@ -34,6 +34,91 @@ public TemplateFunctionalTests(AzureFixture fixture, ITestOutputHelper outputHel
...
+
+                var git = Git(logger, testDirectory);
+                await git.ExecuteAndAssertAsync("init");
+                await git.ExecuteAndAssertAsync($"remote add origin https://{pp.GitUsername}:{pp.GitPassword}@{pp.GitUrl}");
+                await git.ExecuteAndAssertAsync("add .");
+                await git.ExecuteAndAssertAsync("commit -am Initial");
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: This needs quotes around the message right?

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+Microsoft.Extensions.WebEncoders
+microsoft.identitymodel.logging
+microsoft.identitymodel.protocols
+microsoft.identitymodel.protocols.openidconnect
+microsoft.identitymodel.tokens
+microsoft.net.http
+Microsoft.Net.Http.Headers
+Microsoft.Net.Http.Server
+Microsoft.Net.WebSockets.Servern
+newtonsoft.json
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt]: didn't this one have an issue

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt (hunk 4 above)
Comment (pakrym) [reply test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/DotNetCache.txt]: Directory names from actual antares instance

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs (hunk 7 above)
Comment (pakrym) [reply test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/TemplateFunctionalTests.cs]: One word message

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Microsoft.AspNetCore.AzureAppServices.FunctionalTests.csproj [hunk 9]
```diff
@@ -6,7 +6,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <None Include="Templates\*.*" CopyToOutputDirectory="PreserveNewest" />
+    <None Include="Assets\*.*" CopyToOutputDirectory="PreserveNewest" />
+    <Content Include="Assets\DotNetCache.1.0.5.txt">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Microsoft.AspNetCore.AzureAppServices.FunctionalTests.csproj]: Consolidate

File: src/Microsoft.AspNetCore.AzureAppServices.TestBundle/Microsoft.AspNetCore.AzureAppServices.TestBundle.csproj [hunk 10]
```diff
@@ -22,6 +22,7 @@
   <ItemGroup>
     <Content Include="applicationHost.xdt" />
     <Content Include="install.cmd" />
+    <Content Include="dotnet.cmd" />
```
Comment (muratg) [inline src/Microsoft.AspNetCore.AzureAppServices.TestBundle/Microsoft.AspNetCore.AzureAppServices.TestBundle.csproj]: is this required (as opposed to setting the path?)

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Assets/AppServicesWithSiteExtensions.json [hunk 11]
```diff
@@ -29,7 +29,8 @@
       "location": "West US 2",
       "properties": {
         "serverFarmId": "[resourceId(parameters('resourceGroupName'), 'Microsoft.Web/serverFarms', parameters('hostingPlanName'))]"
-      },
+      }
+      ,
```
Comment (muratg) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/Assets/AppServicesWithSiteExtensions.json]: nit: ?

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs [hunk 12]
```diff
@@ -98,8 +98,51 @@ public string GetTimestampedName(string name)
             return name + Timestamp;
         }
 
+        private Dictionary<string, List<Task<IWebApp>>> _deploymentCache = new Dictionary<string, List<Task<IWebApp>>>();
+        private int _predeploymentId;
+
         public async Task<IWebApp> Deploy(string template, IDictionary<string, string> additionalArguments = null, [CallerMemberName] string baseName = null)
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs]: Looks like you don't use baseName anymore

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs [hunk 13]
```diff
@@ -98,8 +98,51 @@ public string GetTimestampedName(string name)
...
         {
+            List<Task<IWebApp>> deployments;
+
+            void DeployBatch()
+            {
+                var maxId = _predeploymentId + 5;
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs]: At the end of our tests we'll have a few unused Web Apps?

File: src/Microsoft.AspNetCore.AzureAppServices.TestBundle/Microsoft.AspNetCore.AzureAppServices.TestBundle.csproj (hunk 10 above)
Comment (pakrym) [reply src/Microsoft.AspNetCore.AzureAppServices.TestBundle/Microsoft.AspNetCore.AzureAppServices.TestBundle.csproj]: We can not set path for git deploy, and directory I'm copying this to is first in PATH so everything fit's nicely.

File: test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs (hunk 13 above)
Comment (pakrym) [reply test/Microsoft.AspNetCore.AzureAppServices.FunctionalTests/AzureFixture.cs]: Yep, I don't think it's too bad but we can add environment variable to limit total number on CI

**Pull request merged**